com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/HDRenderPipeline.VolumetricClouds.cs

```diff
@@ -0,0 +1,684 @@
+            public Texture3D worley128RGBA;
+            public Texture3D worley32RGB;
+            public Texture cloudMapTexture;
+            public Texture cloudLutTexture;
+            public BlueNoise.DitheredTextureSet ditheredTextureSet;
+            public Light sunLight;
+
+            // Compute shader and kernels
+            public ComputeShader volumetricCloudsCS;
+            public int depthDownscaleKernel;
+            public int renderKernel;
+            public int reprojectKernel;
+            public int upscaleAndCombineKernel;
+            public int shadowsKernel;
+
+            // Cloud constant buffer buffer
+            public ShaderVariablesClouds cloudsCB;
+        }
+
+        float Square(float x)
+        {
+            return x * x;
+        }
+
+        float ComputeNormalizationFactor(float earthRadius, float lowerCloudRadius)
+        {
+            return Mathf.Sqrt((earthRadius + lowerCloudRadius) * (earthRadius + lowerCloudRadius) - earthRadius * earthRadius);
+        }
+
+        void GetPresetCloudMapValues(VolumetricClouds.CloudPresets preset, out float densityMultiplier, out float shapeFactor, out float shapeScale, out float erosionFactor, out float erosionScale)
+        {
+            switch (preset)
+            {
+                case VolumetricClouds.CloudPresets.Sparse:
+                {
+                    densityMultiplier = 0.85f;
+                    shapeFactor = 0.95f;
+                    shapeScale = 0.8f;
+                    erosionFactor = 0.5f;
+                    erosionScale = 1.2f;
+                    return;
+                }
+                case VolumetricClouds.CloudPresets.Cloudy:
+                {
+                    densityMultiplier = 0.9f;
+                    shapeFactor = 0.9f;
+                    shapeScale = 0.75f;
+                    erosionFactor = 0.45f;
+                    erosionScale = 1.1f;
+                    return;
+                }
+                case VolumetricClouds.CloudPresets.Overcast:
+                {
+                    densityMultiplier = 0.25f;
+                    shapeFactor = 0.7f;
+                    shapeScale = 1.0f;
+                    erosionFactor = 0.5f;
+                    erosionScale = 1.0f;
+                    return;
+                }
+                case VolumetricClouds.CloudPresets.StormClouds:
+                {
+                    densityMultiplier = 0.3f;
+                    shapeFactor = 0.7f;
+                    shapeScale = 0.875f;
+                    erosionFactor = 0.4f;
+                    erosionScale = 1.0f;
+                    return;
+                }
+            }
+
+            // Default unused values
+            densityMultiplier = 0.6f;
+            shapeFactor = 0.6f;
+            shapeScale = 1.0f;
+            erosionFactor = 0.6f;
+            erosionScale = 1.0f;
+        }
+
+        // The earthRadius
+        const float earthRadius = 6378100.0f;
```

Comment:
note follow the conventoin for constant => k_

Reply:
ok